contracts/dev-utils/contracts/src/OrderValidationUtils.sol

```diff
@@ -0,0 +1,190 @@
+        isValidSignature = EXCHANGE.isValidSignature(
+            orderInfo.orderHash,
+            makerAddress,
+            signature
+        );
+
+        // Get the transferable amount of the `makerAsset`
+        uint256 transferableMakerAssetAmount = getTransferableAssetAmount(order.makerAssetData, makerAddress);
+
+        // Assign to stack variables to reduce redundant mloads
+        uint256 takerAssetAmount = order.takerAssetAmount;
+        uint256 makerFee = order.makerFee;
+    
+        // Get the amount of `takerAsset` that is purchasable given the transferability of `makerAsset` and `makerFeeAsset`
+        uint256 purchasableTakerAssetAmount;
+        if (order.makerAssetData.equals(ZRX_ASSET_DATA)) {
+            // If `makerAsset` equals `makerFeeAsset`, the % that can be filled is
+            // transferableMakerAssetAmount / (makerAssetAmount + makerFee)
+            purchasableTakerAssetAmount = getPartialAmountFloor(
+                transferableMakerAssetAmount,
+                safeAdd(order.makerAssetAmount, makerFee),
+                takerAssetAmount
+            );
+        } else {
+            // Get the transferable amount of the `makerFeeAsset`
+            uint256 transferableMakerFeeAssetAmount = getTransferableAssetAmount(ZRX_ASSET_DATA, makerAddress);
+
+            // If `makerAsset` does not equal `makerFeeAsset`, the % that can be filled is the lower of
+            // (transferableMakerAssetAmount / makerAssetAmount) and (transferableMakerAssetFeeAmount / makerFee)
+            // If `makerFee` is 0, we default to using `transferableMakerAssetAmount`
+            purchasableTakerAssetAmount = makerFee == 0
+                ? getPartialAmountFloor(
+                    transferableMakerAssetAmount,
+                    order.makerAssetAmount,
+                    takerAssetAmount
+                )
+                : min256(
+                    getPartialAmountFloor(
+                        transferableMakerAssetAmount,
+                        order.makerAssetAmount,
+                        takerAssetAmount
+                    ),
+                    getPartialAmountFloor(
+                        transferableMakerFeeAssetAmount,
+                        makerFee,
+                        takerAssetAmount
+                    )
+                );
+        }
+    
+        // `fillableTakerAssetAmount` is the lower of the order's remaining `takerAssetAmount` and the `purchasableTakerAssetAmount`
+        fillableTakerAssetAmount = min256(
+            safeSub(takerAssetAmount, orderInfo.orderTakerAssetFilledAmount),
+            purchasableTakerAssetAmount
+        );
+
+        return (orderInfo, fillableTakerAssetAmount, isValidSignature);
+    }
+
+    /// @dev Fetches information for all passed in orders and the makers/takers of each order.
+    /// @param orders Array of order specifications.
+    /// @param signatures Proofs that orders have been created by makers.
```

Comment:
This sounds a bit awkward. Perhaps just `Order signatures`? Or `Signatures corresponding to the supplied orders`?